src/Runner.Worker/Handlers/ScriptHandler.cs

```diff
@@ -321,12 +321,19 @@ public async Task RunAsync(ActionRunStage stage)
             ExecutionContext.Debug($"{fileName} {arguments}");
 
             Inputs.TryGetValue("standardInInput", out var standardInInput);
-            using (var stdoutManager = new OutputManager(ExecutionContext, ActionCommandManager))
```

Comment:
I think we need to do this for all action invocation, not only for `run` script.

Reply:
Do all step invocations go through the `OutputManager`? Perhaps a cleaner approach for that would be to tie into that class (since it already does the event registration) with the `StalledManager` as a singleton instance. What are your thoughts on that?

Reply:
I believe they do, but we can check the three types
- nodejs actions
- docker actions
- run script steps

Reply:
It seems they all do. Ill rejig it a bit

Reply:
Do we want to start with some doc or ADR to help drive the design conversation?